### ISSUE TITLE
Word insertParagraph is not working on Mac (addin office.js)- only from Developer Window

### ISSUE THREAD

**Issue by danlg** [OfficeDev/office-js#2402]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

# Expected behavior
I am expecting to have text inserted using my dialog. I have spent literally hours and it drives me mad.

# Current behavior
The current behaviour works ONLY from the developer window - WHICH USES THE same code than the DIALOG WINDOW

## Steps to Reproduce

1. Manifest
2. defaults write com.microsoft.Word OfficeWebAddinDeveloperExtras -bool true
2. npm run dev-server and npm run dev in separate window
 The full code included the manifest is here : https://github.com/danlg/bug_addin_react

## Context

* Operating System: Mac OS Big Sur
* Node version: v16.13.2

https://user-images.githubusercontent.com/12525045/151706806-14f94ce0-03fc-4470-8f5e-d03729d5e349.mov

* Office version: 16.57
* Tool version: VS Code 1.63.2

```code
let count = 0
export async function writeToWordImpl (text) {
    return Word.run(async context => {
        count = count +1
        let text_ = text  + count.toString()
        console.log("Writing Word:", text_)
        var body = context.document.body;
        console.log("Writing Word - BODY:", text_)
        // Queue a command to insert the paragraph at the end of the document body.
        try{
            console.log("Writing Word - B4 TRY:", text_)
            body.insertParagraph(text_, Word.InsertLocation.end)
            console.log("Writing Word - SHOULD HAVE INSERTED OK:", text_ )
        }
        catch(error) {
            //never printed
            console.log("Writing Word - CATCH:", error)
        }
        console.log("Writing Word - SHOULD HAVE INSERTED OK 2:", text_)
        // body.insertParagraph('Content of a new paragraph', Word.InsertLocation.end)
        // Synchronize the document state by executing the queued commands,
        // and return a promise to indicate task completion.
        await context.sync().then(function () {
            console.log('Paragraph added at the end of the document body.');
        })
        .catch(function (error) {
            //never printed
            console.log('Error sync paragragh: ', error);
        })
        console.log("Writing to Word 8 - SYNC:", text)
    })
    .catch(function (error) {
        //never printed
        console.log('Error insert paragragh: ', error);
    })
}
```
Manifest is
<Permissions>ReadWriteDocument</Permissions>

**Comment by danlg** [comment]
May be related https://github.com/OfficeDev/office-js/issues/1094 @weshi . Any idea @raevskymichail ?

**Comment by danlg** [comment]
Any idea @grangeryy ?

**Comment by danlg** [comment]
Is someone working on this issue ? This is a blocking point for me to develop a product integrating well with Word and Office applications. What is the alternative @Zlatkovsky, can you coach for 1 hour for a fee?

**Comment by Zlatkovsky** [comment]
@danlg , I've left the Extensibility team some time ago, so I'm afraid I'm no longer involved (nor particularly current) on the product, nor am I doing any coaching/etc. Hopefully someone from the product team can help. You filed your bug in the right place.

FWIW, looking at the video, it's unclear to me what the difference is between the "Developer window" vs "Dialog window". How are you launching one versus the other? I feel like that will be a question that someone from the product team would want to understand.

**Comment by grangeryy** [comment]
@danlg, could you help to provide more details on the differences between "Developer window" and "Dialog window"?

**Comment by danlg** [comment]
Hi @grangeryy, the Developer Window on Mac is launched when the command `defaults write com.microsoft.Word OfficeWebAddinDeveloperExtras -bool true` is run. When the flag is set to false, no Developer Window is appearing, just the Dialog Window appears (but without the possibility to inspect the code and see the console),  this is why I have turned it on. 
My understanding is that the  "Developer window" duplicates the "Dialog window" (it's the same layout) but I could be wrong.

I am following Office official documentation here https://docs.microsoft.com/en-us/office/dev/add-ins/testing/debug-office-add-ins-on-ipad-and-mac.

**Comment by danlg** [comment]
I have published a repo which clearly reproduces the issue https://github.com/danlg/bug_addin_react. I am trying to see in parallel if the same code which is supposed to be cross-platform would run and activate properly the addin Word running on Windows. 
@grangeryy I have updated the case.

**Comment by danlg** [comment]
When running on Windows 10 Entreprise Eval VM, the same code from repo https://github.com/danlg/bug_addin_react, I can compile `npm run dev-server`, but when running `npm run dev`, it hangs on the question "Allow localhost loopback for Microsoft Edge WebView" ? whether I choose y or n.

```[nodemon] clean exit - waiting for changes before restart
Debugging is being started...
App type: desktop
? Allow localhost loopback for Microsoft Edge WebView? (Y/n)
```
I would be very keen to know whether this issue is cross platform or specific to Mac

**Comment by danlg** [comment]
@grangeryy I have updated the case, did you have a chance to have a look at the repo or the issue ? If confirmed, **this bug seems huge (basically removes all interest from an add-in in Word as we cannot interact with the document)** so, in a way, I hope I am doing something wrong and that it can easily be resolved by an expert at Microsoft with some guidance.

**Comment by Zlatkovsky** [comment]
@danlg , from your repro, I think the salient part is that you're trying to call into Word methods from inside a dialog window (launched using `displayDialogAsync`). Per https://docs.microsoft.com/en-us/office/dev/add-ins/develop/dialog-api-in-office-add-ins:

> IMPORTANT!
> - The `messageParent` function is one of *only* two Office JS APIs that can be called in the dialog box.
> - The other JS API that can be called in the dialog box is `Office.context.requirements.isSetSupported`. For information about it, see [Specify Office applications and API requirements](specify-office-hosts-and-api-requirements.md). However, in the dialog box, this API isn't supported in Outlook 2016 one-time purchase (that is, the MSI version).

Essentially, the Office APIs can only be used from within a taskpane, whereas the Dialog API is really only meant for quick modal interactions for things like login windows, etc.

I hope this helps! (And again, I've left the team several years ago... so will let the current members of the team take over form here :-) )

~ Michael

**Comment by yihuax-msft** [comment]
@danlg, @Zlatkovsky is correct. I double checked with our team that API calls from dialog window are not supported, except for MessageParent. We may consider supporting it in the future. Thanks.

**Comment by danlg** [comment]
I have tried to implement using messageParent and, as many complain, the event is not fired for some reason. So this is still blocking and a bug as I am concerned. I have reverted to using the Ribbon instead of a dialog but this is far poorer UX...
